src/factory.rs

```diff
@@ -84,14 +84,14 @@ fn f2i(x: f32) -> I8Norm {
 
 impl Factory {
     #[doc(hidden)]
-    pub fn new(mut backend: BackendFactory, shader_path: &str) -> Self {
+    pub fn new(mut backend: BackendFactory, shader_path: &Path) -> Self {
         let quad_buf = backend.create_vertex_buffer(&QUAD);
         let default_sampler = backend.create_sampler_linear();
         Factory {
             backend: backend,
             scene_id: 0,
             hub: Hub::new(),
-            root_shader_path: shader_path.to_string(),
+            root_shader_path: shader_path.to_owned(),
```

Comment:
If you're going to call `to_owned` then it's probably best to take `PathBuf` instead of `&Path`.

Reply:
Or better still, `Into<PathBuf>`.

Reply:
Actually, it doesn't matter. `WindowBuilder::build` can't transfer own `PathBuf` so it needs either to pass `Path` or clone to the `Renderer::new`. `Renderer::new` uses this `Path` a lot, so `Path` is best choice to avoid cloning. After that, you need to pass it to `Factory::new` - so you need to call `to_owned` now or later - in `Factory::new`, as I did.